### PR TITLE
Cancel vpk install when disk full

### PR DESCRIPTION
**Pull request by R3alCl0ud**
If the disk is full and the vpk isn't already installed, installation fails instead of writing blank files to the disk.

This should fix issue #283.

### CODE REVIEW

File: src/emulator/host/src/app.cpp [hunk 1]
```diff
@@ -176,6 +178,10 @@ bool install_vpk(Ptr<const void> &entry_point, HostState &host, const std::wstri
         } else if (status == UNK_STATE) {
             exit(0);
         }
+    } else if (zip.get()->m_archive_size > boost::filesystem::space(title_base_path).available) {
+        LOG_CRITICAL("Not enough remaining space available to install vpk.");
+        fs::remove_all(title_base_path);
```
Comment (VelocityRa) [inline src/emulator/host/src/app.cpp]: ditto

File: src/emulator/host/src/app.cpp [hunk 2]
```diff
@@ -28,9 +28,9 @@
 #include <kernel/state.h>
 #include <kernel/thread/thread_functions.h>
 #include <util/find.h>
-#include <util/fs.h>
 #include <util/log.h>
 #include <util/string_convert.h>
+#include <boost/filesystem.hpp>
```
Comment (VelocityRa) [inline src/emulator/host/src/app.cpp]: nit: includes from dependencies should go below this line (along with `SDL.h` and `gl.h` below)

File: src/emulator/io/src/io.cpp [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
 #include <io/state.h>
 #include <io/types.h>
-#include <util/fs.h>
+#include <boost/filesystem.hpp>
```
Comment (VelocityRa) [inline src/emulator/io/src/io.cpp]: ditto

**Comment by Zangetsu38** [comment]
@R3alCl0ud Need  rebase and fix conflict

**Comment by Rinnegatamante** [comment]
Bump, any nerws on this?

**Comment by pent0** [comment]
Boing! This is start to get outdated with master branch. Any news?

**Comment by R3alCl0ud** [comment]
Been busy with school and family things (family member passed away). I can work on this once the semester is over (two weeks) but you can close this for now if you are going to do that IO rewrite that was planned along time ago.

**Comment by 1whatleytay** [comment]
Bump?

**Pull request closed without merging**